File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    endDays = []
    
    for i in range(len(progresses)):
        endDay = (100 - progresses[i]) // speeds[i]
        
        if (100 - progresses[i]) % speeds[i] != 0:
            endDay += 1
        
        endDays.append(endDay)
        
    answer = [0]
    tempDay = endDays[0]
    for d in endDays:
        if tempDay >= d:
            answer[-1] += 1
        else:
            answer.append(1)
            tempDay = d
            
    return answer

File: 프로그래머스/lv2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let releases = [];
    
    let prevNum = -1;
    let releaseCnt = 1;
    for (let i = 0; i < progresses.length; i++) {
        let progress = progresses[i];
        let speed = speeds[i];
        
        let due = Math.ceil((100 - progress) / speed);
        
        if(i === 0) {
            prevNum = due;
            continue;
        }
        
        if (prevNum >= due) {
            releaseCnt++;
        } else {
            prevNum = due;
            releases.push(releaseCnt);
            releaseCnt = 1;
        }
    }
    
    if (releaseCnt != 0) releases.push(releaseCnt);
    
    return releases;
}